cmds/bmp/update_state_command_test.go

```diff
@@ -69,13 +74,81 @@ var _ = Describe("update-state command", func() {
 			Expect(validate).To(BeTrue())
 			Expect(err).ToNot(HaveOccurred())
 		})
+
+		Context("when validating a bad UpdateStateCommand", func() {
+			Context("when no server ID or state is passed", func() {
+				BeforeEach(func() {
+					options = cmds.Options{
+						Verbose: false,
+					}
+				})
+
+				It("fails validation with errors", func() {
+					cmd = bmp.NewUpdateStateCommand(options, fakeBmpClient)
+					validate, err := cmd.Validate()
+					Expect(validate).To(BeFalse())
+					Expect(err).To(HaveOccurred())
+				})
+			})
+
+			Context("when server ID isn't passed", func() {
+				BeforeEach(func() {
+					options = cmds.Options{
+						Verbose: false,
+						Server:  "",
+						State:   "bm.state.new",
+					}
+				})
+
+				It("fails validation with errors", func() {
+					cmd = bmp.NewUpdateStateCommand(options, fakeBmpClient)
+					validate, err := cmd.Validate()
+					Expect(validate).To(BeFalse())
+					Expect(err).To(HaveOccurred())
+				})
+			})
+
+			Context("when state isn't passed", func() {
+				BeforeEach(func() {
+					options = cmds.Options{
+						Verbose:  false,
+						Server:   "fake-server-id",
+						Password: "bm.state.new",
+					}
+				})
+
+				It("fails validation with errors", func() {
+					cmd = bmp.NewUpdateStateCommand(options, fakeBmpClient)
+					validate, err := cmd.Validate()
+					Expect(validate).To(BeFalse())
+					Expect(err).To(HaveOccurred())
+				})
+			})
+
+			Context("when state isn't valid", func() {
+				BeforeEach(func() {
+					options = cmds.Options{
+						Verbose:  false,
+						Server:   "fake-server-id",
+						Password: "fake-state",
+					}
+				})
+
+				It("fails validation with errors", func() {
+					cmd = bmp.NewUpdateStateCommand(options, fakeBmpClient)
+					validate, err := cmd.Validate()
+					Expect(validate).To(BeFalse())
+					Expect(err).To(HaveOccurred())
+				})
+			})
+		})
 	})
 
 	Describe("#Execute", func() {
 		It("executes a good UpdateStateCommand", func() {
 			rc, err := cmd.Execute(args)
-			Expect(rc).To(Equal(0))
-			Expect(err).ToNot(HaveOccurred())
+			Expect(rc).To(Equal(1))
```

Comment:
Yes. Right. This is just for passing Travis temporarily. And I tried to mock user input here to do UT for Execute(), as it need an user confirmation(https://github.com/cloudfoundry-community/bosh-softlayer-tools/pull/50/files#diff-c1126fe3cf047c635c322ffa4e661b5aR70), but it got nothing. Could you please help shed some light on UT for this command? Thank you:)